packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -293,31 +293,31 @@ class CupertinoPageTransition extends StatelessWidget {
     @required Animation<double> primaryRouteAnimation,
     @required Animation<double> secondaryRouteAnimation,
     @required this.child,
-    bool linearTransition,
-  }) :
-      _primaryPositionAnimation = linearTransition
-        ? _kRightMiddleTween.animate(primaryRouteAnimation)
-        : _kRightMiddleTween.animate(
-            new CurvedAnimation(
-              parent: primaryRouteAnimation,
-              curve: Curves.easeOut,
-              reverseCurve: Curves.easeIn,
-            )
-          ),
-      _secondaryPositionAnimation = _kMiddleLeftTween.animate(
-        new CurvedAnimation(
-          parent: secondaryRouteAnimation,
-          curve: Curves.easeOut,
-          reverseCurve: Curves.easeIn,
-        )
-      ),
-      _primaryShadowAnimation = _kGradientShadowTween.animate(
-        new CurvedAnimation(
-          parent: primaryRouteAnimation,
-          curve: Curves.easeOut,
-        )
-      ),
-      super(key: key);
+    @required bool linearTransition,
```

Comment:
I'd just default to false instead of requiring it

Reply:
It's a private constructor and we already specify it everywhere we use it, so a default value wouldn't matter.